src/azure-cli/azure/cli/command_modules/acs/decorator.py

```diff
@@ -2149,6 +2171,269 @@ def get_network_policy(self, enable_validation: bool = False, **kwargs) -> Union
+        """Obtain the value of enable_addons.
+
+        Note: enable_addons will not be decorated into the `mc` object.
+
+        This function supports the option of enable_split. When enabled, it will split the string into a list with
+        "," as the delimiter.
+
+        :return: string, empty list or list of strings, or None
+        """
+        # read the original value passed by the command
+        enable_addons = self.raw_param.get("enable_addons")
+
+        if kwargs.get("enable_split", False):
+            enable_addons = enable_addons.split(',') if enable_addons else []
+
+        # this parameter does not need validation
+        return enable_addons
+
+    # pylint: disable=unused-argument
+    def get_workspace_resource_id(self, **kwargs) -> Union[str, None]:
+        """Dynamically obtain the value of workspace_resource_id according to the context.
+
+        When both workspace_resource_id is not assigned, dynamic completion will be triggerd. Function
+        "_ensure_default_log_analytics_workspace_for_monitoring" will be called to create a workspace with
+        subscription_id and resource_group_name.
+
+        This function supports the option of read_only. When enabled, it will skip dynamic completion and validation.
+
+        :return: string or None
+        """
+        # read the original value passed by the command
+        workspace_resource_id = self.raw_param.get("workspace_resource_id")
+        # try to read the property value corresponding to the parameter from the `mc` object
+        read_from_mc = False
+        if (
+            self.mc and
+            self.mc.addon_profiles and
+            CONST_MONITORING_ADDON_NAME in self.mc.addon_profiles and
+            self.mc.addon_profiles.get(
+                CONST_MONITORING_ADDON_NAME
+            ).config.get(CONST_MONITORING_LOG_ANALYTICS_WORKSPACE_RESOURCE_ID)
+        ):
+            workspace_resource_id = self.mc.addon_profiles.get(
+                CONST_MONITORING_ADDON_NAME
+            ).config.get(CONST_MONITORING_LOG_ANALYTICS_WORKSPACE_RESOURCE_ID)
+            read_from_mc = True
+
+        # skip dynamic completion & validation if option read_only is specified
+        if kwargs.get("read_only"):
+            return workspace_resource_id
+
+        # dynamic completion
+        if not read_from_mc:
+            if workspace_resource_id is None:
+                # use default workspace if exists else create default workspace
+                workspace_resource_id = (
+                    _ensure_default_log_analytics_workspace_for_monitoring(
+                        self.cmd,
+                        self.intermediates.get("subscription_id", None),
+                        self.get_resource_group_name(),
+                    )
+                )
+            # normalize
+            workspace_resource_id = workspace_resource_id.strip()
+            if not workspace_resource_id.startswith("/"):
```

Comment:
use "/" + workspace_resource_id.strip("/")?

Reply:
Nice catch, changed in commit [7924eed](https://github.com/Azure/azure-cli/pull/19478/commits/7924eedf40a8a71d37bad0a1b377c5f0589a0877) with a little bit more modification.